spec/lib/orbf/rules_engine/builders/calculator_factory_spec.rb

```diff
@@ -75,6 +75,39 @@ def solve_score_table(value)
           expect(solution["my_var"]).to be_between(1, 4)
         end
       end
+
+      describe 'eval_array' do
+        it 'solves' do
+          solution = calculator.solve("my_var" => "sum(eval_array('a', ARRAY(1,3,4), 'b', ARRAY(2,4,5), 'b - a'))")
+          expect(solution["my_var"]).to eq(1+1+1)
+        end
+
+        describe 'raises on invalid input' do
+          it 'not same length arrays' do
+            expect{
+              calculator.solve!("my_var" => "sum(eval_array('a', ARRAY(), 'b', ARRAY(1), 'b - a'))")
+            }.to raise_error(StandardError)
```

Comment:
I would assert on the message (standard error can be quiet common ;))
and as the message will show up to the user, would make sure they are meaningful